### PR TITLE
docs: Application sets metrics documentation

### PR DESCRIPTION
**Pull request by kostis-codefresh**
Documentation for https://github.com/argoproj/argo-cd/pull/19691

Checklist:

* [X] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [X] The title of the PR states what changed and the related issues number (used for the release note).
* [X] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [X] Does this PR require documentation updates?
* [X] I've updated documentation as required by this PR.
* [X] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [X] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [X] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

**Review by reggie-k — APPROVED** [review]

File: docs/operator-manual/metrics.md [hunk 1]
```diff
@@ -83,6 +86,23 @@ The example below will expose the Argo CD Application condition `OrphanedResourc
       - ExcludedResourceWarning
 ```
 
+## Application Set Controller metrics
+
+The Application Set controller exposes the following metrics for application sets.
+
+| Metric | Type | Description |
+|--------|:----:|-------------|
+| `argocd_appset_info` | gauge | Information about Application Sets. It contains labels for name and namespace of an application set as well as `Resource_update_status`  that reflects the `ResourcesUpToDate` property |
```
Comment (todaywasawesome) [inline docs/operator-manual/metrics.md]:
```suggestion
| `argocd_appset_info` | gauge | Information about Application Sets. It contains labels for the name and namespace of an application set as well as `Resource_update_status`  that reflects the `ResourcesUpToDate` property |
```

File: docs/operator-manual/metrics.md [hunk 2]
```diff
@@ -83,6 +86,23 @@ The example below will expose the Argo CD Application condition `OrphanedResourc
...
+| `argocd_appset_reconcile` | histogram | Application reconciliation performance in seconds. It contains labels for name and namespace of an applicationset |
```
Comment (todaywasawesome) [inline docs/operator-manual/metrics.md]:
```suggestion
| `argocd_appset_reconcile` | histogram | Application reconciliation performance in seconds. It contains labels for the name and namespace of an applicationset |
```

File: docs/operator-manual/metrics.md [hunk 3]
```diff
@@ -83,6 +86,23 @@ The example below will expose the Argo CD Application condition `OrphanedResourc
...
+| `argocd_appset_labels` | gauge | Applicationset labels translated to Prometheus labels. Disabled by default |
+| `argocd_appset_owned_applications` | gauge | Number of applications owned by the applicationset. It contains labels for name and namespace of an applicationset. |
+
+Similar to the same metric in application controller (`argocd_app_labels`) the metric `argocd_appset_labels` is disabled by default. You can enable it by providing the `–metrics-applicationset-labels` argument to the application set controller.
```
Comment (todaywasawesome) [inline docs/operator-manual/metrics.md]:
```suggestion
Similar to the same metric in application controller (`argocd_app_labels`) the metric `argocd_appset_labels` is disabled by default. You can enable it by providing the `–metrics-applicationset-labels` argument to the applicationset controller.
```

**Review by todaywasawesome — CHANGES_REQUESTED** [review]
Made a few small grammar suggestions for you to review @kostis-codefresh .

File: docs/operator-manual/metrics.md [hunk 4]
```diff
@@ -83,6 +86,23 @@ The example below will expose the Argo CD Application condition `OrphanedResourc
...
+| `argocd_appset_reconcile` | histogram | Application reconciliation performance in seconds. It contains labels for the name and namespace of an applicationset |
+| `argocd_appset_labels` | gauge | Applicationset labels translated to Prometheus labels. Disabled by default |
+| `argocd_appset_owned_applications` | gauge | Number of applications owned by the applicationset. It contains labels for the name and namespace of an applicationset. |
+
+Similar to the same metric in application controller (`argocd_app_labels`) the metric `argocd_appset_labels` is disabled by default. You can enable it by providing the `–metrics-applicationset-labels` argument to the applicationset controller.
+
+Once enabled it works exactly the same as application controller metrics (label_ appended to normalized label name).
+Available labels include Name, Namespace + all labels enabled by the command line options and their value (exactly like application controller metrics described in the previous section).
+
+
```
Comment (nitishfy) [inline docs/operator-manual/metrics.md:105]:
```suggestion

```

Comment (nitishfy) [reply docs/operator-manual/metrics.md:105]: nit

**Review by nitishfy — APPROVED** [review]

**Review by wanghong230 — APPROVED** [review]
Thank you for documenting those things well.

**Review by alexmt — APPROVED** [review]

**Review by ishitasequeira — APPROVED** [review]

**Review by ishitasequeira — COMMENTED** [review]
@todaywasawesome are you good with the changes you requested?

**Review by alexmt — APPROVED** [review]
Thank you.

**Review by todaywasawesome — APPROVED** [review]
Great!

**Pull request merged**